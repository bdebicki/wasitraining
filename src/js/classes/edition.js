'use strict';

export class edition {
	constructor(editionData) {
		this._id = editionData.id;
		this._year = editionData.editionYear;
		this._place = editionData.place;
		this._name = editionData.fullName;
		this._headliners = editionData.headliners;
		this._details = editionData.details;
	}

	get editionId() {
		return this._id;
	}

	get editionYear() {
		return this._year;
	}

	get editionFullName() {
		return this._name;
	}

	get editionPlace() {
		return `${this._place.object}, ${this._place.city}`;
	}

	get editionDetails() {
		return this._details;
	}

	get editionLength() {
		return this.editionDetails.length;
	}

	get headliners() {
		return this._headliners;
	}

	get editionDate() {
		const firstDay = this.editionDetails[0].day;

		if(this.editionLength > 1) {
			const lastDay =  this.editionDetails[this.editionLength - 1].day;
			return {firstDay, lastDay};
		} else {
			return {firstDay};
		}
	}

	get editionRain() {
		for(let day of this.editionDetails) {
			if(day.rain === true) {
				return 'yes';
			}
		}

		return 'no';
	}
}