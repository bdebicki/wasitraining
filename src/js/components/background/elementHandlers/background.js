import LAYOUT from '../../../elementHandlers/layout';

const bgClassBuilders = {
	cover: LAYOUT.BG_COVER_ID,
	video: LAYOUT.BG_VIDEO_ID,
};
const BG = {
	COVER_CLASS: `${bgClassBuilders.cover}__cover`,
	VIDEO_CLASS: `${bgClassBuilders.video}__video`,
};

export { BG as default };
