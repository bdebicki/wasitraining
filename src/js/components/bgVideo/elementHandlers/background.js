import LAYOUT from '../../../elementHandlers/layout';

const bgClassBuilder = LAYOUT.BG_VIDEO_ID;
const BG = {
  VIDEO_CLASS: `${bgClassBuilder}__video`,
};

export { BG as default };
