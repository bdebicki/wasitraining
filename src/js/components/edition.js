'use strict';

export class edition {
	constructor(data) {
		this.id = data.id;
		this.edition = data.edition;
		this.place = data.place;
		this.dates = data.dates;
	}

	get editionId() {
		return this.id;
	}

	get editionName() {
		return this.edition;
	}

	get editionPlace() {
		return this.place;
	}

	get editionLength() {
		return this.dates.length;
	}

	get editionDates() {
		const firstDay = this.editionDates()[0].day;
		const lastDay =  this.editionDates()[this.editionLength - 1].day;
		return {firstDay, lastDay};
	}

	get editionDays() {
		return this.dates;
	}
}