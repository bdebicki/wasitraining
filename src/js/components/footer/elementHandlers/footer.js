const aboutClassBuilder = 'aboutSection';
const FOOTER = {
	COOKIES_CLASS: 'cookiesSection',
	ABOUT_SECTION_CLASS: aboutClassBuilder,
	COPYRIGHTS_CLASS: `${aboutClassBuilder}__copyrights`,
	ABOUT_LINK_CLASS: `${aboutClassBuilder}__aboutLink`,
};

export { FOOTER as default };
