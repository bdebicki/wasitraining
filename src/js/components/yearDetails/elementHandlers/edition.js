const editionClassBuilders = {
	details: 'editionDetails',
	lineup: 'shortLineup',
	headliner: 'headliner',
	headlines: 'headliners',
};
const EDITION = {
	EDITION_DETAILS_ID: editionClassBuilders.details,
	EDITION_DETAILS_YEAR_CLASS: `${editionClassBuilders.details}--edition`,
	FULL_NAME_CLASS: `${editionClassBuilders.details}__fullName`,
	PLACE_CLASS: `${editionClassBuilders.details}__place`,
	YEAR_CLASS: `${editionClassBuilders.details}__year`,
	DATES_CLASS: `${editionClassBuilders.details}__date`,
	LINEUP_CLASS: editionClassBuilders.lineup,
	LINEUP_EDITION_CLASS: `${editionClassBuilders.lineup}--edition`,
	HEADLINERS_CLASS: `${editionClassBuilders.lineup}__${editionClassBuilders.headlines}`,
	HEADLINER_CLASS: editionClassBuilders.headliner,
	HEADLINERS_DAY_CLASS: `${editionClassBuilders.headliner}--day`,
	LINEUP_LINK_CLASS: `${editionClassBuilders.lineup}__detailsLink`,
};

export {
	editionClassBuilders,
	EDITION,
};
