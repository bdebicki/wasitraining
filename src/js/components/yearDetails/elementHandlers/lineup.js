const lineupClassBuilder = {
  section: 'lineup',
  artists: 'lineupArtists',
  line: 'lineupArtists__line',
  lvl: 'lineupArtists__lvl',
  artist: 'lineupArtists__artist',
  slice: 'lineupArtists__slice',
  multipleArtists: 'lineupArtists__multipleArtists',
};
const LINEUP = {
  SECTION_ID: lineupClassBuilder.section,
  EDITION_CLASS: `${lineupClassBuilder.section}--edition`,
  ARTISTS_CLASS: lineupClassBuilder.artists,
  ARTISTS_EDITION_CLASS: `${lineupClassBuilder.artists}--edition`,
  ARTISTS_DAY_CLASS: `${lineupClassBuilder.artists}__day`,
  ARTISTS_NEW_LINE_ELEMENT_CLASS: `${lineupClassBuilder.artists}__newLine`,
  ARTISTS_SEPARATOR_ELEMENT_CLASS: `${lineupClassBuilder.artists}__separatorElement`,
  ARTISTS_LINE_CLASS: lineupClassBuilder.line,
  ARTISTS_LEVEL_CLASS: lineupClassBuilder.lvl,
  ARTIST_CLASS: lineupClassBuilder.artist,
  ARTIST_MULTIPLE_ARTISTS_CLASS: lineupClassBuilder.multipleArtists,
  ARTIST_MULTIPLE_ARTISTS_ARTIST_CLASS: `${lineupClassBuilder.multipleArtists}__artist`,
  ARTIST_SLICE_CLASS: `${lineupClassBuilder.slice}`,
};

export {
  lineupClassBuilder,
  LINEUP,
};
