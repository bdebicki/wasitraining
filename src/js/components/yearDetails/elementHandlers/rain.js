import globalClassBuilders from '../../../elementHandlers/globalClassBuilders';

const rainClassBuilders = {
  section: 'rainSection',
  info: 'rainInfo',
  header: 'rainHeader',
  details: 'rainDetails',
  detailsList: 'rainDetailsList',
};
const RAIN = {
  SECTION_ID: rainClassBuilders.section,
  INFO_CLASS: rainClassBuilders.info,
  INFO_YES_CLASS: `${rainClassBuilders.info}--yes`,
  INFO_NO_CLASS: `${rainClassBuilders.info}--no`,
  MASK_CLASS: `${rainClassBuilders.info}__mask`,
  YES_SHAPE_CLASS: `${rainClassBuilders.info}__maskShape--yes`,
  NO_SHAPE_CLASS: `${rainClassBuilders.info}__maskShape--no`,
  VIDEO_CLASS: `${rainClassBuilders.info}__video`,
  YES_VIDEO_CLASS: `${rainClassBuilders.info}__video--yes`,
  NO_VIDEO_CLASS: `${rainClassBuilders.info}__video--no`,
  HEADER_CLASS: rainClassBuilders.header,
  HEADER_HIDDEN_CLASS: `${rainClassBuilders.header}--${globalClassBuilders.hidden}`,
  HEADLINE_CLASS: `${rainClassBuilders.header}__headline`,
  DETAILS_LINK_CLASS: `${rainClassBuilders.details}__link`,
  CLOSE_DETAILS_CLASS: `${rainClassBuilders.details}__close`,
  DETAILS_ID: rainClassBuilders.details,
  DETAILS_CLASS: rainClassBuilders.details,
  DETAILS_LIST_CLASS: rainClassBuilders.detailsList,
  DETAILS_ITEM_CLASS: `${rainClassBuilders.detailsList}__item`,
  DETAILS_ITEM_DAY_CLASS: `${rainClassBuilders.detailsList}__day`,
  DETAILS_ITEM_RAIN_CLASS: `${rainClassBuilders.detailsList}__rain`,
};

export { RAIN as default };
