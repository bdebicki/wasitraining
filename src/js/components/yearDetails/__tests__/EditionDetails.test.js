import cleanDOM from '../../../../../tests/utils/cleanDOM';
import editionOneDay from '../../../../../tests/__mocks__/editionOneDay.json';
import editionMultipleDays from '../../../../../tests/__mocks__/edition-notSort-noMergeArtists.json';
import editionNewEdition from '../../../../../tests/__mocks__/edition-notSort-customLevels.json';
import { EDITION } from '../elementHandlers/edition';
import EditionDetails from '../EditionDetails';

describe('yeaar details - edition details', () => {
	afterAll(() => cleanDOM());

	const details = new EditionDetails(editionMultipleDays);

	describe('return edition date', () => {
		it('return date for one day edition', () => {
			// when
			const oneDayDetails = new EditionDetails(editionOneDay);

			// then
			expect(oneDayDetails.decorateEditionDates()).toBe('2nd July');
		});

		it('return date for multiple days edition', () => {
			// then
			expect(details.decorateEditionDates()).toBe('3rd July - 6th July');
		});
	});

	describe('render code parts', () => {
		it('renders headliners section', () => {
			expect(details.renderHeadliners()).toMatchSnapshot();
		});

		it('renders edition details section', () => {
			expect(details.renderEditionDetails()).toMatchSnapshot();
		});

		it('renders composed edition details section', () => {
			expect(details.render()).toMatchSnapshot();
		});
	});

	describe('update actions', () => {
		// having
		document.body.appendChild(details.render());
		const newEdition = new EditionDetails(editionNewEdition);

		newEdition.updateEditionDetails();
		it('update edition details', () => {
			// when
			newEdition.updateEditionDetails();

			// then
			expect(document.getElementById(EDITION.EDITION_DETAILS_ID)).toMatchSnapshot();
		});
	});
});
