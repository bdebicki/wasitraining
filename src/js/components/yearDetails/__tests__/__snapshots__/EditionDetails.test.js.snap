// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`yeaar details - edition details render code parts renders composed edition details section 1`] = `
<section
  class="editionDetails--edition2013"
  data-year="2013"
  id="editionDetails"
>
  <h2
    class="editionDetails__year"
  >
    2013
  </h2>
  <h3
    class="editionDetails__fullName"
  >
    Heineken sponsors Open'er Festival
  </h3>
  <p
    class="editionDetails__date"
  >
    3rd July - 6th July
  </p>
  <p
    class="editionDetails__place"
  >
    Babie Doły Airport, Gdynia
  </p>
  <div
    class="shortLineup shortLineup--edition2013"
  >
    <ul
      class="shortLineup__headliners"
    >
      <li
        class="headliners__headliner"
      >
        Blur
      </li>
      <li
        class="headliners__headliner"
      >
        Arctic Monkeys
      </li>
      <li
        class="headliners__headliner"
      >
        Queens of the Stone Age
      </li>
      <li
        class="headliners__headliner"
      >
        Kings of Leon
      </li>
    </ul>
    <p
      class="shortLineup__detailsLink"
    >
      <a
        class="link link--inverted link--sizeXS link--hasIcon"
        href="#lineup"
      >
        see full lineup
        <svg
          class="icon--plus link__icon"
          height="5px"
          viewBox="0 0 5 5"
          width="5px"
        >
          <rect
            height="1"
            width="5"
            x="0"
            y="2"
          />
          <rect
            height="5"
            width="1"
            x="2"
            y="0"
          />
        </svg>
      </a>
    </p>
  </div>
</section>
`;

exports[`yeaar details - edition details render code parts renders edition details section 1`] = `
<DocumentFragment>
  <h2
    class="editionDetails__year"
  >
    2013
  </h2>
  <h3
    class="editionDetails__fullName"
  >
    Heineken sponsors Open'er Festival
  </h3>
  <p
    class="editionDetails__date"
  >
    3rd July - 6th July
  </p>
  <p
    class="editionDetails__place"
  >
    Babie Doły Airport, Gdynia
  </p>
</DocumentFragment>
`;

exports[`yeaar details - edition details render code parts renders headliners section 1`] = `
<ul
  class="shortLineup__headliners"
>
  <li
    class="headliners__headliner"
  >
    Blur
  </li>
  <li
    class="headliners__headliner"
  >
    Arctic Monkeys
  </li>
  <li
    class="headliners__headliner"
  >
    Queens of the Stone Age
  </li>
  <li
    class="headliners__headliner"
  >
    Kings of Leon
  </li>
</ul>
`;

exports[`yeaar details - edition details update actions update edition details 1`] = `
<section
  class="editionDetails--edition2004"
  data-year="2004"
  id="editionDetails"
>
  <h2
    class="editionDetails__year"
  >
    2004
  </h2>
  <h3
    class="editionDetails__fullName"
  >
    Heineken Music sponsors Open'er Festival
  </h3>
  <p
    class="editionDetails__date"
  >
    2nd July - 3rd July
  </p>
  <p
    class="editionDetails__place"
  >
    Kościuszki Square, Gdynia
  </p>
  <div
    class="shortLineup shortLineup--edition2004"
  >
    <ul
      class="shortLineup__headliners"
    >
      <li
        class="headliners__headliner"
      >
        Pink
      </li>
      <li
        class="headliners__headliner"
      >
        Cypress Hill
      </li>
      <li
        class="headliners__headliner"
      >
        Massive Attack
      </li>
      <li
        class="headliners__headliner"
      >
        Goldfrapp
      </li>
    </ul>
    <p
      class="shortLineup__detailsLink"
    >
      <a
        class="link link--inverted link--sizeXS link--hasIcon"
        href="#lineup"
      >
        see full lineup
        <svg
          class="icon--plus link__icon"
          height="5px"
          viewBox="0 0 5 5"
          width="5px"
        >
          <rect
            height="1"
            width="5"
            x="0"
            y="2"
          />
          <rect
            height="5"
            width="1"
            x="2"
            y="0"
          />
        </svg>
      </a>
    </p>
  </div>
</section>
`;
